### PR TITLE
POC Reducing Cassandra query count for GetMailboxes

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java [hunk 1]
```diff
@@ -117,21 +121,24 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
...
     }
 
-    private Stream<Mailbox> retrieveSpecificMailboxes(MailboxSession mailboxSession, ImmutableList<MailboxId> mailboxIds) {
+    private Stream<Mailbox> retrieveSpecificMailboxes(MailboxSession mailboxSession, ImmutableList<MailboxId> mailboxIds) throws MailboxException {
+        Quotas quotas = quotaLoader.getQuotas(mailboxSession.getUser());
         return mailboxIds
             .stream()
             .map(mailboxId -> mailboxFactory.builder()
                 .id(mailboxId)
+                .withUserQuota(quotas)
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java]: do you loose per mailbox quota, or did I miss somethnig?

Comment (chibenwa) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java]: In JMAP, yes

Comment (rouazana) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java]: which quota ratio API? As far as I understand, Inbox choose to only refer to INBOX's quota to know about the ratio. But until their there is a different quota on each mailbox.

Comment (rouazana) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java]: and what about shared mailboxes?

**Comment by chibenwa** [comment]
200 user doing getmailboxes during 30 mins. O**Only have default mailboxes** so the impact on prod like data will be greater.

master: 

capture d ecran de 2019-02-01 15-54-22

This branch:

capture d ecran de 2019-02-01 15-54-25

Conclusions:
 - p 99 divided by 7 (!!!)
 - mean time divided by 5
 - throughput increase is minimal due to long sleep delays between calls (1 to 2 s)

**Comment by chibenwa** [comment]
Subpart of this work were already extracted and merged

**Pull request closed without merging**